nx-X11/programs/Xserver/hw/nxagent/Keystroke.c

```diff
@@ -249,6 +249,11 @@ static void parse_keystroke_file(void)
   char *homefile = "/.nx/config/keystrokes.cfg";
   char *etcfile = "/etc/nxagent/keystrokes.cfg";
 
+  if (nxagentKeystrokeFileParsed)
+    return;
+
+  nxagentKeystrokeFileParsed = True;
```

Comment:
Don't set this to true, if the parsing has not happened so far.

It actually should only be set to true, if the parsing of one of the potential files has _really_ been successful.